script.rb

```diff
@@ -0,0 +1,112 @@
+# rubocop:disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    my_new_arr = []
+    my_each { |i| my_new_arr.push i if (yield i) == true }
+    my_new_arr
+  end
+
+  def my_all?(arg = nil)
+    if arg
+      my_each { |i| return false unless arg.is_a?(Class) && i.is_a?(arg) || arg.is_a?(Regexp) && i.match(arg) }
+    elsif block_given?
+      my_each { |i| return false unless (yield i) == true }
+    else
+      my_each { |i| return false if i == false || i.nil? }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if arg
+      my_each { |i| return true if arg.is_a?(Class) && i.is_a?(arg) || arg.is_a?(Regexp) && i.match(arg) }
+    elsif block_given?
+      my_each { |i| return true if (yield i) == true }
+    else
+      my_each { |i| return true if i }
+    end
+    false
+  end
+
+  def my_none?(arg = nil)
+    if arg
+      my_each { |i| return false if arg.is_a?(Class) && i.is_a?(arg) || arg.is_a?(Regexp) && i.match(arg) }
+    elsif block_given?
+      my_each { |i| return false if (yield i) == true }
+    else
+      my_each { |i| return false if i }
+    end
+    true
+  end
```

Comment:
## 3. #my_none?
 when a pattern other than Regex or a Class is given, `my_none?` should return false only if one of the collection matches the pattern